services/iplant-groups/src/iplant_groups/clients/grouper.clj

```diff
@@ -714,16 +714,15 @@
 (defn- remove-existing-permissions
   [username attribute-def-name & [params]]
   (->> (get-permission-assign-ids username attribute-def-name params)
-       (map (fn [[permission-type ids]] (remove-permission-assign-ids username permission-type ids)))
-       dorun))
+       (mapv (fn [[permission-type ids]] (remove-permission-assign-ids username permission-type ids)))))
 
-(defn- remove-existing-role-permissions
```

Comment:
It appears you used the membership version of this outside of this file, but not the role version. I'm ambivalent on whether it'd make sense to add the parallel endpoints now for consistency, make this private for consistency, or neither, so your call but something to consider.